addons/mail/models/mail_channel.py

```diff
@@ -579,6 +579,10 @@ def channel_info(self, extra_info = False):
                                           .channel_partner_ids
                                           .filtered(lambda p: p.id != self.env.user.partner_id.id)
                                           .read(['id', 'name', 'im_status']))
+                for direct_partner in info['direct_partner']:
+                    users = self.env['res.partner'].browse(direct_partner['id']).user_ids
```

Comment:
I just noticed that we are reading user_ids and out_of_office_message in a loop. It is also the case for the read(['id', 'name', 'im_status']. I guess that we could extract this logic out of the loop and simplifies hr_holidays channel_info.